crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -262,6 +262,27 @@ def create_proposal
     base
   end
 
+  def disable_non_core_proposals
+    # Find all non-core proposals and remove all roles that belong
+    # to those proposals from the nodes
+    active_non_core_roles = RoleObject.find_roles_by_name("*-config-*").reject(&:core_role?)
+    active_non_core_roles.each do |role|
+      @logger.debug("deactivating proposal #{role.name}")
+
+      roles_to_remove = role.elements.keys
+      roles_to_remove << role.name
+
+      roles_to_remove.each do |delete_role|
+        nodes = NodeObject.find("roles:#{delete_role}")
+        nodes.each do |node|
+          node.delete_from_run_list(delete_role)
+          node.save
```

Comment:
Just thinking: aren't we saving node(s) quite often this way? Wouldn't it be better to iterate over all nodes, for each one delete all roles that need to be removed and than save the node only once?

(But maybe it's not relevant or node.save is so fast that we do not need to bother...)